Add a public Filters property to IRequest

Jericho opened the pull request:
Resolves #42

Jericho committed:
Add a public Filters property to IRequest

Jericho committed:
No need to convert collection to array when creating the request

Pathoschild commented on Client/FluentClient.cs:
```diff
@@ -96,7 +96,7 @@ public virtual IRequest SendAsync(HttpRequestMessage message)
             // clone the underlying message because HttpClient doesn't normally allow re-sending
             // the same request, which would break IRequestCoordinator.
-            return new Request(message, this.Formatters, request => this.BaseClient.SendAsync(request.Message.Clone(), request.CancellationToken), this.Filters.ToArray())
+            return new Request(message, this.Formatters, request => this.BaseClient.SendAsync(request.Message.Clone(), request.CancellationToken), this.Filters)
```
This makes the request filters a reference to the client filters, so changes to the request filters will unexpectedly be global. For example:

```cs
// one client filter
Console.WriteLine(client.Filters.Length);

// remove request filter
IRequest request = await client.GetAsync(…);
request.Filters.Remove<DefaultErrorFilter>();
IResponse response = await request;

// zero client filters
Console.WriteLine(client.Filters.Length);

Jericho replied:
You're right, the request filters is now a reference to client filters. I'll fix. I removed `ToArray` because collection becomes fixed in size which prevents adding/removing items. I'll use `ToList` instead.

Jericho committed:
Make a copy of the client filters instead of using a reference.

This is important to ensure that adding/removing a filter from the request does not remove modify the client filters.

Jericho committed:
Formatting

Pathoschild commented:
Merged into `develop` for the upcoming 3.0 release. Thanks!
